src/parse/parser.rs

```diff
@@ -0,0 +1,522 @@
+            _ => e,
+        });
+        // returns high-level AST or Error
+        match try!(querytype) {
+
+            // Different Query-types are matched and returned
+
+            // Create-Query
+            Keyword::Create => {
+                let query = Query::DefStmt(DefStmt::Create(try!(self.parse_create_stmt())));
+                return Ok(try!(self.return_query_ast(query)));
+            },
+
+            // Alter-Query
+            Keyword::Alter => {
+                let query = Query::DefStmt(DefStmt::Alter(try!(self.parse_alt_stmt())));
+                return Ok(try!(self.return_query_ast(query)));
+            },
+
+            // Drop-Query
+            Keyword::Drop => {
+                let query = Query::DefStmt(DefStmt::Drop(try!(self.parse_drop_stmt())));
+                return Ok(try!(self.return_query_ast(query)));
+            },
+
+            // Use-Query
+            Keyword::Use => {
+                let query = Query::ManipulationStmt(ManipulationStmt::Use(try!(self.parse_use_stmt())));
+                return Ok(try!(self.return_query_ast(query)));
+            }
+            // Unknown Error
+            _ => return Err(ParseError::UnknownError)
+        }
+    }
+
+
+
+/*+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#Parser Functions
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*/
+
+
+
+    // Starts the parsing for tokens in create-syntax
+    fn parse_create_stmt(&mut self) -> Result<CreateStmt, ParseError> {
+        // Convention: Every method must use bump to
+        // put the lexer to the position of the token the method needs
+        self.bump();
+
+        match try!(self.expect_keyword(&[Keyword::Table,Keyword::Database])) {
+            // Create the table subtree
+            Keyword::Table => return Ok(CreateStmt::Table(try!(self.parse_create_table_stmt()))),
+            // Create Database subtree
+            Keyword::Database => { self.bump();
+             return Ok(CreateStmt::Database(try!(self.expect_word())));
+         },
+            // Create the view subtree
+            // Keyword::View => ...
+
+            // Unknown parsing error
+            _ => return Err(ParseError::UnknownError),
+        };
+
+    }
+
+    // Parses the tokens fore the create table subtree
+    fn parse_create_table_stmt(&mut self) -> Result<CreateTableStmt, ParseError> {
+        // Convention: Every method must use bump to
+        // put the lexer to the position of the token the method needs
+        self.bump();
+
+        // create a CreateTableStmt Object with the table id
+        let mut table_info = CreateTableStmt { tid: try!(self.expect_word()), cols: Vec::<ColumnInfo>::new() };
+
+        self.bump();
+        // if there is a ParenOp token.....
+        if self.curr.is_none() {
+            return Ok(table_info);
+        }
```

Comment:
agree for all that want to know why
if needs to return early cause it does not have an else

this would probably work
```Rust
if self.curr.is_none() {
    Ok(table_info)
} else {

    try!(self.expect_token(&[Token::ParenOp]));

    table_info.cols = try!(self.parse_create_column_vec());
    Ok(table_info)
}
```